code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/LdapTokenCreator.java

```diff
@@ -39,10 +39,7 @@ public Token getToken(ApiRequest request) {
             throw new ClientVisibleException(ResponseCodes.SERVICE_UNAVAILABLE, "LdapConfig", "LdapConfig is not Configured.", null);
         }
         String code = ObjectUtils.toString(requestBody.get(SecurityConstants.CODE));
-        String[] split = code.split(":");
-        if (split.length != 2) {
-            throw new ClientVisibleException(ResponseCodes.FORBIDDEN);
-        }
+        String[] split = code.split(":", 2);
```

Comment:
Do you still want to check for length of two? You could end up with an ArrayIndexOutOfBounds on the next line.